pwem/protocols/protocol_filter_by_normal.py

```diff
@@ -0,0 +1,141 @@
+    CHOICES = [
+        CHOICE_SIDE_VIEW,
+        CHOICE_TOP_VIEW
+    ]
+
+    # --------------------------- DEFINE param functions ----------------------
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+        form.addParam('input', PointerParam, pointerClass=EMSet, label='Input',
+                      help='Set which items will be filtered.')
+        form.addParam('selection', EnumParam, label='Selection',
+                      choices=self.CHOICES, default=0,
+                      help='Select which views are kept. Side views keep items '
+                           'with tilt angles near 90 or 270 degrees. Top '
+                           'views keep items with tilt angles near 0 or 180 '
+                           'degrees.')
+        form.addParam('tolerance', FloatParam, label='Tolerance (deg)',
+                      default=15.0, validators=[Range(0.0, 90.0)],
+                      help='Select the tolerance used for keeping elements in '
+                           'degrees. This defines the deviation from the angle '
+                           'mentioned in the selection. Must be in [0, 90] '
+                           'range.')
+        form.addParam('transformField', StringParam, label='Transform field',
+                      default='_transform', expertLevel=LEVEL_ADVANCED,
+                      help='Field used to retrieve the transform matrix of '
+                      'the items.')
+
+    # --------------------------- INSERT steps functions ----------------------
+    def _insertAllSteps(self):
+        self._insertFunctionStep(self.createOutputStep)
+
+    # --------------------------- STEPS functions -----------------------------
+    def createOutputStep(self):
+        inputSet = self._getInputSet()
+        outputSet = inputSet.createCopy(self._getPath(), copyInfo=True)
+        
+        ANGLE_TARGETS = [90.0, 0.0]
+        target = math.radians(ANGLE_TARGETS[self.selection.get()])
+        tolerance = math.sin(math.radians(self.tolerance.get()))
+        for item in inputSet:
+            matrix = self._getMatrix(item)
+            _, tilt, _ = euler_from_matrix(matrix, 'szyz')
+            tilt -= target
+            s = abs(math.sin(tilt))
+            if s < tolerance:
+                outputSet.append(item)
+            
+        self._defineOutputs(**{ProtSetFilterByNormalOutputs.Output.name: outputSet})
+        self._defineSourceRelation(self.input, outputSet)
+            
+    # --------------------------- INFO functions ------------------------------
+    def _validate(self):
+        errors = []
+
+        inputSet = self._getInputSet()
+        item = inputSet.getFirstItem()
+        try:
+            self._getMatrix(item)
+            
+        except (AttributeError, TypeError, ValueError):
+            errors.append('Input items must have a valid transform associated. '
+                          'Try modifying "Matrix Field" parameter if necessary')
```

Comment:
I guess, that not all items have _transform, thus the field "Matrix Field".  Wouldn't be better to scan attributes to find the transform one... otherwise, how the user is goingto find the attribute name?